### PR TITLE
implement hff heuristic

### PR DESCRIPTION
**Pull request by tomsilver**
on oracle, compared to hadd:
* very slightly worse for cover
* slightly better for blocks
* significantly worse for painting
* significantly better for playroom (@amburger66 please verify)

### CODE REVIEW

**Comment by amburger66** [comment]
verified! tried a handful of combinations of open/closed doors, and all were solvable :0 very cool

File: src/utils.py [hunk 1]
```diff
@@ -1014,6 +1022,52 @@ def _accumulate(distances: Collection[float]) -> float:
...
 
 
+class _HFFHeuristic(_RelaxationHeuristic):
+    """Implements the HFF delete relaxation heuristic.
+    """
+    @staticmethod
+    def _accumulate(distances: Collection[float]) -> float:
+        return sum(distances)
+
+    def calc_goal_h(self) -> float:
```
Comment (ronuchit) [inline src/utils.py:1032]: is this copied from pyperplan?

File: src/settings.py [hunk 2]
```diff
@@ -57,7 +57,7 @@ class GlobalSettings:
     random_options_max_tries = 100
 
     # SeSamE parameters
-    task_planning_heuristic = "hadd"  # hadd or hmax
+    task_planning_heuristic = "hadd"  # hadd or hmax or hff
```
Comment (ronuchit) [inline src/settings.py:60]: based on your comment, let's just make this env-specific for the sake of progress? i'd be fine with that

Comment (ronuchit) [reply src/settings.py:60]: predicate invention envs are all using hadd, while amber project is using hff, so it's not an issue paper-wise

**Comment by tomsilver** [comment]
@ronuchit stop working! you're on vacation!

File: src/utils.py (hunk 1 above)
Comment (tomsilver) [reply src/utils.py:1032]: mostly yeah, a few very minor changes for style

File: src/settings.py (hunk 2 above)
Comment (tomsilver) [reply src/settings.py:60]: sounds good

**Pull request merged**